src/components/richText/imageNode/ImageNode.tsx

```diff
@@ -0,0 +1,233 @@
+import {
+  $applyNodeReplacement,
+  DOMConversionMap,
+  DOMConversionOutput,
+  DOMExportOutput,
+  EditorConfig,
+  LexicalEditor,
+  createEditor,
+  DecoratorNode,
+  LexicalNode,
+  NodeKey,
+  SerializedEditor,
+  SerializedLexicalNode,
+  Spread,
+} from "lexical";
+
+import * as React from "react";
+
+import LexicalImage from "./LexicalImage";
+
+/**
+ * The raw Object obtained when first parsing a serialized {@link ImageNode}.
+ */
+export type SerializedImageNode = Spread<
+  {
+    altText: string;
+    caption: SerializedEditor;
+    height?: number;
+    maxWidth: number;
+    showCaption: boolean;
+    src: string;
+    width?: number;
+    type: "image";
+    version: 1;
+  },
+  SerializedLexicalNode
+>;
+
+/**
+ * Defines a Lexical Dev {@link DecoratorNode} for images. These images can optionally contain captions.
+ */
+export class ImageNode extends DecoratorNode<JSX.Element> {
+  __src: string;
+  __altText: string;
+  __width: "inherit" | number;
+  __height: "inherit" | number;
+  __maxWidth: number;
+  __showCaption: boolean;
+  __caption: LexicalEditor;
+  // Captions cannot yet be used within editor cells
+  __captionsEnabled: boolean;
+
+  constructor(
+    src: string,
+    altText: string,
+    maxWidth: number,
+    width?: "inherit" | number,
+    height?: "inherit" | number,
+    showCaption?: boolean,
+    caption?: LexicalEditor,
+    captionsEnabled?: boolean,
+    key?: NodeKey
+  ) {
+    super(key);
+    this.__src = src;
+    this.__altText = altText;
+    this.__maxWidth = maxWidth;
+    this.__width = width || "inherit";
+    this.__height = height || "inherit";
+    this.__showCaption = showCaption || false;
+    this.__caption = caption || createEditor();
+    this.__captionsEnabled = captionsEnabled || captionsEnabled === undefined;
```

Comment:
nit: can we remove the `|| captionsEnabled === undefined` since `captionsEnabled` would already be undefined if it's not specified?

Reply:
Removed caption support! They weren't using it in Entity Edit as it turns out.